### ISSUE TITLE
Plot Colors don't match with the color bar

### ISSUE THREAD

**Issue by haideralipunjabi** [geopandas/geopandas#1603]
I am trying to plot a map of my state with some data. Each district of the state has a value associated with it (except for 2 districts), and I am trying to show those values using a chloropleth map.

Here's the data (in list format):  

    In:      print(list(data['value']))
    Out:    [31, 49, 14, 57, 27, 22, 137, 64, 33, 8, 28, 17, 15, 6, 10, 14, 484, 21, 16, 26, None, None]

Here's my plot code:

    ax1=data.plot(column='value',legend=True,ax=ax,cmap=color,edgecolor="black",linewidth=0.2, missing_kwds={
          "linewidth":0.2,
          "edgecolor":"black",
          "facecolor":"white"
     })

I am not using the colorbar from `legend` parameter (it's not producing the colorbar for some reason, but the legend it produces is good for debugging).

To make the colorbar, I use the following:

     norm = Normalize(vmin=data['value'].min(),vmax=data['value'].max())
     n_cmap = cm.ScalarMappable(norm=norm, cmap=color)
     n_cmap.set_array([])
     divider = make_axes_locatable(ax1)
     cax = divider.new_vertical(size="2.5%", pad=0.1, pack_start=True)
     fig.add_axes(cax)
     fig.colorbar(n_cmap,label=name,cax=cax, orientation="horizontal")

It produces a colorbar, but it doesn't match the colors plotted on the map.

[![enter image description here][1]][1]

As you can see in the image, the colorbar isn't matching the map (or it's legend). I believe that the colorbar is accurate as 137 should be yellowish while only one entry/district should be red (the 484 one). 

I did further debugging by messing around with `plotting.py` file from geopandas. For some reason, the `categorical` boolean turns True at line 644 because `values.dtype is np.dtype("O")` is `True`
and it changes the `values` that I passed to something else, causing the `vmin` and `vmax` to become `0` & `18` (instead of `6` & `484`), at which point everything starts going over my head, and I realised I should ask here.

*TLDR: I want the colors in the map to match the colors in the colorbar (the colorbar colors are accurate).*

  [1]: https://i.stack.imgur.com/NWTt5.png

**Comment by haideralipunjabi** [comment]
Really sorry, I was initializing `data['value']` to `None` instead of `numpy.nan`. A good night's sleep does miracles.
